### PR TITLE
Fix flake8 and tox dependencies

### LINKED ISSUE #196: Pre-commit hooks failing
Our Pre-Commit CI service is throwing errors about some of the hooks.

## Task

- [ ] fix code errors related to pre-commit

### LINKED ISSUE #197: flake8 version mismatch when running `poetry install`
When running `poetry install` I get the following error.

> Resolving dependencies... (0.8s)
>
> Because no versions of flake8 match >5.0.4,<6.0.0
 and flake8 (5.0.4) depends on importlib-metadata (>=1.1.0,<4.3), flake8 (>=5.0.4,<6.0.0) requires importlib-metadata (>=1.1.0,<4.3).
And because tox (3.14.0) depends on importlib-metadata (>=0.12,<1), flake8 (>=5.0.4,<6.0.0) is incompatible with tox (3.14.0).
So, because django-invitations depends on both tox (3.14.0) and flake8 (^5.0.4), version solving failed.

When running `poetry add --group dev flake8`, I get the following error.

> Resolving dependencies... (0.8s)
>
> Because no versions of flake8 match >5.0.4,<6.0.0
 and flake8 (5.0.4) depends on importlib-metadata (>=1.1.0,<4.3), flake8 (>=5.0.4,<6.0.0) requires importlib-metadata (>=1.1.0,<4.3).
And because tox (3.14.0) depends on importlib-metadata (>=0.12,<1), flake8 (>=5.0.4,<6.0.0) is incompatible with tox (3.14.0).
So, because django-invitations depends on both tox (3.14.0) and flake8 (^5.0.4), version solving failed.

## Task

- [ ] resolve Poetry dependency errors

### PR DESCRIPTION
**Pull request by brylie**
Closes #196
Closes #197

Changes

- use latest published versions of `flake8` and `tox`

References

- https://pypi.org/project/flake8/#history
- https://pypi.org/project/tox/#history

### CODE REVIEW

**Comment by brylie** [comment]
I am converting this to draft since `pre-commit` is causing some hassle. I am also disabling auto fix on `pre-commit` since it is causing issues with local branch sync.

**Comment by brylie** [comment]
OK, this is ready for review. Hoever, I'm unsure why the test coverage is dropping since most changes are lint.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -19,25 +20,22 @@ repos:
...
   - repo: https://github.com/pycqa/flake8
     rev: 5.0.4
     hooks:
       - id: flake8
         additional_dependencies:
           - flake8-bugbear
           - flake8-comprehensions
-          - flake8-tidy-imports
         args: [--max-line-length=88]
-  - repo: https://github.com/asottile/reorder_python_imports
```
Comment (brylie) [inline .pre-commit-config.yaml:31]: Removing this since it is already handled by `isort`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -52,12 +50,4 @@ repos:
   - repo: https://github.com/asottile/yesqa
     rev: v1.4.0
     hooks:
-      -   id: yesqa
-  - repo: https://github.com/asottile/add-trailing-comma
```
Comment (brylie) [inline .pre-commit-config.yaml:56]: Removing this since it is handled by `black`.

**Comment by brylie** [comment]
@jezdez, could you take a look at this PR?

**Review by jezdez — APPROVED** [review]
LGTM!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (valberg) [reply .pre-commit-config.yaml:31]: This is not entirely true - isort does not seperate imports into multiple lines: https://github.com/asottile/reorder_python_imports#why-this-style

I vote we replace isort with reorder_pyton_imports.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (valberg) [reply .pre-commit-config.yaml:56]: Not entirely. Black does not add trailing commas when the arguments include *, *args or **kwargs.

I vote we keep add-trailing-comma

Comment (brylie) [reply .pre-commit-config.yaml:56]: Ok, we can add it in a follow-up PR

**Pull request merged**